packages/core/src/services/health/service.ts

```diff
@@ -466,14 +466,43 @@ export class Service {
         conditions: EErrorConditions.NO_RESPONSE,
       };
     }
+    let referenceUrls: IReferenceURL[];
+    const { healthyOracles, badOracles } = await this.getOracles(chain);
+    if (healthyOracles.length >= 1) {
+      referenceUrls = healthyOracles;
+    } else {
+      const healthyPeers = await this.getPeers(chain, node.id);
+      if (!healthyOracles.length && healthyPeers.length < 2) {
+        return {
+          ...healthResponse,
+          status: EErrorStatus.ERROR,
+          conditions: EErrorConditions.NO_PEERS,
+        };
+      } else if (!healthyOracles.length && healthyPeers.length >= 2) {
+        healthResponse = {
+          ...healthResponse,
+          details: { noOracle: true, numPeers: healthyPeers.length },
+        };
+      }
+      referenceUrls = [...healthyOracles, ...healthyPeers];
+    }
+
+    if (badOracles.length) {
+      healthResponse = {
+        ...healthResponse,
+        details: {
+          ...healthResponse.details,
+          badOracles: badOracles.map(({ url }) => url),
+        },
+      };
+    }
 
-    let referenceUrl = 'https://rpc.mainnet.near.org';
 
     try {
       /* Get node's block height, highest block height from reference nodes */
       const [internalBh, externalBh] = await Promise.all([
         this.getNEARBlockHeight(url),
-        this.getNEARBlockHeight(referenceUrl),
+        this.getNEARBlockHeight(referenceUrls[0].url),
```

Comment:
Generally we want to be using the highest block height from the referenceUrls array, similar to how we do for EVM. So I would suggest either checking all of the block heights for the reference URL and selecting the highest or - if you think just one block height is sufficient - = modifying the logic above that creates the referenceUrl array to something along the lines of `healthyOracles[0] || healthyPeers[0]`